modules/dnn/src/cuda4dnn/primitives/convolution.hpp

```diff
@@ -272,6 +272,36 @@ namespace cv { namespace dnn { namespace cuda4dnn {
                 }
             }
 
+            if (rank == 3)
+            {
+                // Conv1D
+                // We add extra dim for input tensor, because CuDNN doesn't support convolution with 3D tensors
+                std::size_t input_width = params.input_shape.back();
+                params.input_shape.pop_back();
+                params.input_shape.push_back(1);
+                params.input_shape.push_back(input_width);
```

Comment:
I don't have strong opinions about this but I think doing a `pop_back` and then two `push_back`s is more confusing/convoluted than a single `insert` call (which also makes the intention explicit). I think this  three line sequence can be written more succinctly as:

```
auto& input_shape = params.input_shape;
input_shape.insert(std::begin(input_shape), 1);
```